addons/web/static/src/legacy/legacy_service_provider.js

```diff
@@ -3,16 +3,21 @@
 import { bus } from "web.core";
 import { browser } from "../core/browser/browser";
 import { registry } from "../core/registry";
+import { handleDoAction } from "./action_adapters";
 
 export const legacyServiceProvider = {
-    dependencies: ["effect"],
+    dependencies: ["effect", "action"],
     start({ services }) {
         browser.addEventListener("show-effect", (ev) => {
             services.effect.add(ev.detail.type, ev.detail);
         });
         bus.on("show-effect", this, (payload) => {
             services.effect.add(payload.type, payload);
         });
+
+        browser.addEventListener("do-action", (ev) => {
+            handleDoAction(services.action, ev.detail);
```

Comment:
I'm a bit concerned that we might loose behavior because the `this.onReverseBreadcrumb = ...` line is not there in case of the activity systray items. I think aab said this feature was no longer relevant but I'm not sure.